### DISCUSSION TITLE
Mention in the documentation where to create config.json

### DISCUSSION THREAD

**Discussion by olegkrivtsov** [renovatebot/renovate#11636] [Request Help]
### What would you like Renovate to be able to do?

While familiarizing myself with Renovate I installed it on my Ubuntu VM and tried `yarn start`. I saw a message saying that I need to add the personal access token. OK, but where to add it? I checked the documentation (the section named Configuration), and it said I should modify `config.json`. I checked the root directory of rennovate, but there was no config.json there, and the documentation had no mention if I should create it myself or take some of its dist versions.

I also found some config file in the dist/config directory, and I was curious if those are some templates. I found out those are some base config file I can extend. So it was a bit tricky for me.

Suggestion:

- In the documentation section https://docs.renovatebot.com/self-hosted-configuration/ (or in Getting Started), mention that you should create `config.json` yourself
- Provide an example of that `config.json` file to start with in the documentation.

For example, I came to this config.json eventually, but I'm still curious if it was correct or not:

```
{
  "extends": ["config:base"],
  "username":"self-hosted-renovate[bot]",
  "gitAuthor":"Self-hosted Renovate Bot <123456+self-hosted-renovate[bot]@users.noreply.github.enterprise.com>",
  "token":"x-access-token:XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
  "repositories": ["YYYYYYYYY/ZZZZZZZ"]
}
```

### If you have any ideas on how this should be implemented, please tell us here.

Please extend the documentation in the relevant places to simplify config.json creation for newbies.

### Is this a feature you are interested in implementing yourself?

Yes

**Comment by viceice** [comment]
Did you read this [docs/development/local-development.md](https://github.com/renovatebot/renovate/blob/HEAD/docs/development/local-development.md) ?

**Reply by olegkrivtsov** [reply]
Hi @viceice sure, I read it. It has only two mentions of config.js, and none of them show how to create it. Also, that document is intended for contributors, but I thought of those users who just wanted to run Renovate locally as I did.

**Reply by viceice** [reply]
`yarn start` command is for develeopment / debugging only. It's not recomend to use for production usage.

See other answer

**Comment by rarkins** [comment]
It's not mandatory to create a `config.json` (we usually say `config.js` btw - are you sure it said `config.json`)?

We could perhaps add more details or at least a link in this section:
![image](https://user-images.githubusercontent.com/6311784/132473820-157e37fb-28b0-4f58-8588-4ea91711debc.png)

From here: https://docs.renovatebot.com/self-hosted-configuration/

**Reply by olegkrivtsov** [reply]
@rarkins yes, It says `config.js`: https://docs.renovatebot.com/getting-started/running/

"Global config

Renovate's server-side/admin config is referred to as its "global" config, and can be specified using either a config file (config.js or config.json), environment variables, or CLI parameters."

**Reply by viceice** [reply]
just put it in current directory (will be ignored by git when testing from renovate source
